inst/include/sampler.h

```diff
@@ -403,7 +413,71 @@ namespace smc {
     {
         return expl(2*stableLogSumWeights(pPopulation.GetLogWeight())-stableLogSumWeights(2.0*pPopulation.GetLogWeight()));
     }
+    template <class Space, class Params>
+    std::vector<population<Space>> sampler<Space,Params>::GetApop(void) const
+    {
+        long N = GetNumber();
+        long lHistorylength = GetHistoryLength();
+        int historylength = static_cast<int>(lHistorylength);

```

Comment:
Is there a reason to keep casting long to int? In the abstract losing bits like this seems a bad idea, although `long` and `int` may not be different on some architectures.